Wikipedia/Code/TalkPageContainerViewController.swift

```diff
@@ -92,9 +92,22 @@ class TalkPageContainerViewController: ViewController, HintPresenting {
     var fromNavigationStateRestoration: Bool = false
     private var cancellationKey: String?
     
-    private var currentLoadingViewController: ViewController?
-    private var currentSourceView: UIView?
-    private var currentSourceRect: CGRect?
+    //If populated, talk page will automatically route to matching reply thread after topic list loads
+    private var sectionTitleFragment: String?
+    
+    //Overlay and activity indicator that displays while routing to reply thread
+    private lazy var replyRoutingOverlay: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = theme.colors.paperBackground
+        return view
+    }()
+    private lazy var replyRoutingActivityIndicator: UIActivityIndicatorView = {
+        let view = UIActivityIndicatorView(style: .medium)
+        view.color = theme.isDark ? .white : .gray
```

Comment:
This same color convention is already applied in other places so please push back if it doesn't make sense, but wouldn't it be preferable to just use `theme.colors.primaryText`, so the indicator always renders with the same readability as text?

Reply:
@staykids that does make sense, I'm not sure why it's been done in this pattern in the past. Maybe holdover from the style having incorporated only white and gray options? I made this change.